File: src/main/java/com/example/AI_Study_Planer/dto/response/Quiz/QuizQuestionResponse.java
package com.example.AI_Study_Planer.dto.response.Quiz;

import com.example.AI_Study_Planer.enums.Quiz.QuestionType;
import lombok.Data;

import java.util.List;

@Data
public class QuizQuestionResponse {

    private String id;

    private String questionText;

    private QuestionType type;

    private Integer order;

    private List<QuestionOptionResponse> options;
}

File: src/types/Response/Quiz/QuizQuestionResponse.ts
import type { QuestionType } from "@/types/Enums";
import type { QuizOptionResponse } from "./QuizOptionResponse";

export interface QuizQuestionResponse {
  id: string;

  questionText: string;

  type: QuestionType;

  order: number;

  options: QuizOptionResponse[];
}
